tools/lint/black.py

```diff
@@ -72,6 +70,9 @@ def main():
     args = parser.parse_args()
     init_environment(args)
 
+    logging.info("Installing black (if needed)...")
+    run_command(["pip", "install", "-q", "black"], check=True)
```

Comment:
```suggestion
    run_command(["pipx", "install", "-q", "black"], check=True)
```

Ubuntu 24.04 (which I believe is what this workflow is running on) complains [1] when using `pip` to install packages. `pipx` might be a better option. But it also needs to be installed first.

[https://stackoverflow.com/questions/75608323/how-do-i-solve-error-externally-managed-environment-every-time-i-use-pip-3]